### PR TITLE
Add table aws_appautoscaling_target. closes #329

### LINKED ISSUE #329: Add table aws_appautoscaling_target
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/applicationautoscaling

### PR DESCRIPTION
**Pull request by Priyanka585464**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_appautoscaling_target []

PRETEST: tests/aws_appautoscaling_target

TEST: tests/aws_appautoscaling_target
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_dynamodb_table.example: Creating...
aws_dynamodb_table.example: Still creating... [10s elapsed]
aws_dynamodb_table.example: Creation complete after 18s [id=turbottest22904]
aws_appautoscaling_target.named_test_resource: Creating...
aws_appautoscaling_target.named_test_resource: Creation complete after 3s [id=table/turbottest22904]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = 986325076436
aws_partition = aws
aws_region = us-east-1
resource_id = table/turbottest22904
resource_name = turbottest22904

Running SQL query: test-get-query.sql
[
  {
    "partition": "aws",
    "region": "us-east-1",
    "resource_id": "table/turbottest22904"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "partition": "aws",
    "region": "us-east-1",
    "resource_id": "table/turbottest22904"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "region": "us-east-1",
    "title": "table/turbottest22904"
  }
]
✔ PASSED

POSTTEST: tests/aws_appautoscaling_target

TEARDOWN: tests/aws_appautoscaling_target

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  service_namespace,
  scalable_dimension,
  resource_id,
  creation_time
from
  aws_appautoscaling_target
where
  service_namespace = 'dynamodb';
```
```
+-------------------+----------------------------------+------------------------------------------+---------------------+
| service_namespace | scalable_dimension               | resource_id                              | creation_time       |
+-------------------+----------------------------------+------------------------------------------+---------------------+
| dynamodb          | dynamodb:table:ReadCapacityUnits | table/turbot-test-20200155-create-update | 2021-04-27 08:11:37 |
+-------------------+----------------------------------+------------------------------------------+---------------------+
```

### Check whether autoscaling is enabled of both read-write capacity

```sql
select
  resource_id,
  scalable_dimension
from
  aws_appautoscaling_target
where
  service_namespace = 'dynamodb'
  and scalable_dimension = 'dynamodb:table:ReadCapacityUnits'
  or 'scalable_dimension' = 'dynamodb:table:WriteCapacityUnits';
```
```
+------------------------------------------+----------------------------------+
| resource_id                              | scalable_dimension               |
+------------------------------------------+----------------------------------+
| table/turbot-test-20200155-create-update | dynamodb:table:ReadCapacityUnits |
+------------------------------------------+----------------------------------+
```

### CODE REVIEW

File: aws/service.go [hunk 1]
```diff
@@ -140,6 +141,27 @@ func AutoScalingService(ctx context.Context, d *plugin.QueryData, region string)
 	return svc, nil
 }
 
+// ApplicationAutoScalingService returns the service connection for AWS ApplicationAutoScaling service
```
Comment (Subhajit97) [inline aws/service.go:144]:
```suggestion
// ApplicationAutoScalingService returns the service connection for AWS Application Auto Scaling service
```

File: aws/table_ aws_appautoscaling_target.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+	"github.com/aws/aws-sdk-go/service/applicationautoscaling"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsAppAutoScalingTarget(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_appautoscaling_target",
+		Description: "AWS ApplicationAutoScaling Target",
```
Comment (Subhajit97) [inline aws/table_ aws_appautoscaling_target.go:19]:
```suggestion
		Description: "AWS Application Auto Scaling Target",
```

File: aws/table_ aws_appautoscaling_target.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
+package aws
```
Comment (Subhajit97) [inline aws/table_ aws_appautoscaling_target.go:1]: Fix file name, there is a whitespace after `table_`.

File: aws/table_ aws_appautoscaling_target.go [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+func getAwsApplicationAutoScalingTarget(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAwsApplicationAutoScalingTarget")
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+
+	name := d.KeyColumnQuals["service_namespace"].GetStringValue()
+	ID := d.KeyColumnQuals["resource_id"].GetStringValue()
```
Comment (Subhajit97) [inline aws/table_ aws_appautoscaling_target.go:129]: don't use uppercase when defining any local variables, use lowerCamelCase instead. 
```suggestion
	id := d.KeyColumnQuals["resource_id"].GetStringValue()
```

File: aws/table_ aws_appautoscaling_target.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+		ServiceNamespace: &name,
+		ResourceIds:      []*string{types.String(ID)},
+	}
+
+	// Get call
+	op, err := svc.DescribeScalableTargets(params)
+	if err != nil {
+		return nil, err
+	}
+	if len(op.ScalableTargets) > 0 {
```
Comment (Subhajit97) [inline aws/table_ aws_appautoscaling_target.go:148]:
```suggestion
	if op.ScalableTargets != nil && len(op.ScalableTargets) > 0 {
```

File: docs/tables/aws_appautoscaling_target.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+```sql
+select
+  resource_id,
+  scalable_dimension
+from
+  aws_appautoscaling_target
+where
+  service_namespace = 'dynamodb'
+  and scalable_dimension = 'dynamodb:table:ReadCapacityUnits'
+  or 'scalable_dimension' = 'dynamodb:table:WriteCapacityUnits';
```
Comment (Subhajit97) [inline docs/tables/aws_appautoscaling_target.md:33]:
```suggestion
  or scalable_dimension = 'dynamodb:table:WriteCapacityUnits';
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@Priyanka585464 There are few changes required. Please see comments, thanks!

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
@Priyanka585464 can we identify below 

1. Identify and list empty AWS Auto Scaling Groups (ASGs).
2. Ensure that logging is enabled in the ASGs
3. Ensure AWS Auto Scaling Groups utilize multiple Availability Zones to improve environment reliability.

**Comment by Priyanka585464** [comment]
@rajmohanty17 
The query examples you have suggested I think those are related to autoscaling group resource maybe .The response what i am getting from get and list call , i am afraid that we can form these queries out of those response:
```
[
 {
 "account_id": "986325076436",
 "creation_time": "2021-04-27 08:11:37",
 "max_capacity": 100,
 "min_capacity": 5,
 "partition": "aws",
 "region": "us-east-1",
 "resource_id": "table/turbot-test-20200155-create-update",
 "role_arn": null,
 "scalable_dimension": "dynamodb:table:ReadCapacityUnits",
 "service_namespace": "dynamodb",
 "suspended_state": {
  "DynamicScalingInSuspended": false,
  "DynamicScalingOutSuspended": false,
  "ScheduledScalingSuspended": false
 },
 "title": "table/turbot-test-20200155-create-update"
 }
]
```
If you have any additional information regarding the suggested queries, please let me know.

**Comment by rajmohanty17** [comment]
For the time being let's proceed with it, will revisit later as per requirement.

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**